src/iconv.jl

```diff
@@ -1,9 +1,77 @@
 # This file is a part of Julia. License is MIT: http://julialang.org/license
 
 module iconv
-import Base: close, eof, flush, read, readall, write
+import Base: close, eof, flush, read, readall, write, show
 import Base.Libc: errno, strerror, E2BIG, EINVAL, EILSEQ
 export StringEncoder, StringDecoder, encode, decode
+export StringEncodingError, GeneralError
+export InvalidEncodingError, InvalidSequenceError, IncompleteSequenceError, IConvError
+
+
+abstract StringEncodingError
+
+# For now, don't display module name, maybe want it if this changes to StringEncoders
+const modulename = ""
+
+# specified encodings are not supported
+type InvalidEncodingError <: StringEncodingError
+    mod::ByteString
+    msg::ByteString
+    args
+    InvalidEncodingError(from, to) = new(modulename, "Conversion from <<1>> to <<2>> not supported by iconv implementation, check that specified encodings are correct", (from, to))
+end
+
+# Encountered invalid byte sequence
+type InvalidSequenceError <: StringEncodingError
+    mod::ByteString
+    msg::ByteString
+    args
+    InvalidSequenceError(seq) = new(modulename, "Byte sequence 0x<<1>> is invalid in source encoding or cannot be represented in target encoding", (seq,))
+end
+
+# Input ended with incomplete byte sequence
+type IncompleteSequenceError <: StringEncodingError
+    mod::ByteString
+    msg::ByteString
+    args
+    IncompleteSequenceError() =
+	new(modulename, "Incomplete byte sequence at end of input", ())
+end
+
+type IConvError <: StringEncodingError
+    mod::ByteString
+    msg::ByteString
+    args
+    IConvError(func) = new(func, "<<1>> (<<2>>)", (errno(), strerror(errno())))
+end
+
+type OutputBufferError <: StringEncodingError
+    mod::ByteString
+    msg::ByteString
+    args
+    OutputBufferError(func) =
+	new(modulename, "Ran out of space in the output buffer", ())
+end
+
+type InvalidBytesError <: StringEncodingError
+    mod::ByteString
+    msg::ByteString
+    args
+    InvalidBytesError() =
+	new(modulename, "Invalid byte sequence in input", ())
+end
+
+iconv_error(func) = throw(IConvError(func))
```

Comment:
I must have misunderstood, I thought you had asked me to change `generalerror` to `iconv_error`, and add `IConvError`.
I was just trying to do what you asked, I'll change that.